<launch>
  <arg name="uav_name" default="$(optenv UAV_NAME uav1)"/>

  <arg name="standalone" default="false"/>

  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg uav_name)_uvdar_nodelet_manager"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>

  <!-- Node Settings -->
  <arg name="output" default="screen"/>
  <arg name="proc" default="false"/>
  <arg name="calib" default="false"/>

  <arg name="debug" default="false"/>
  <arg name="visual_debug" default="false"/>
  <arg name="gui" default="true"/>
  <arg name="frame" default="$(arg uav_name)/local_origin_stable"/>
  <arg name="filterCount" default="4"/>
  <arg name="useVelocity" default="false"/>
  <arg name="outputFramerate" default="60"/>

  <arg name="indoor" default="false"/>
  <arg name="odometry_available" default="true"/>

  <group ns="$(arg uav_name)">


  
  <!-- <node pkg="uvdar" type="uvdar_kalman_node" name="filterUvdar" output="screen" launch-prefix="debug_roslaunch"> -->
  <node pkg="uvdar" type="uvdar_kalman_node" name="filterUvdar" output="screen">
    <param name="output_frame" value="$(arg frame)"/>
    <param name="useVelocity" value="$(arg useVelocity)" />
    <param name="outputFramerate" value="$(arg outputFramerate)" />

    <param name="indoor" value="$(arg indoor)" />
    <param name="odometry_available" value="$(arg odometry_available)" />

    <param name="filterCount" value="$(arg filterCount)" />

    <remap from="measuredPose1" to="/$(arg uav_name)/uvdar/measuredPose1"/>
    <remap from="measuredPose2" to="/$(arg uav_name)/uvdar/measuredPose2"/>
    <remap from="measuredPose3" to="/$(arg uav_name)/uvdar/measuredPose3"/>
    <remap from="measuredPose4" to="/$(arg uav_name)/uvdar/measuredPose4"/>

    <remap from="filteredPose1" to="/$(arg uav_name)/uvdar/filteredPose1"/>
    <remap from="filteredPose2" to="/$(arg uav_name)/uvdar/filteredPose2"/>
    <remap from="filteredPose3" to="/$(arg uav_name)/uvdar/filteredPose3"/>
    <remap from="filteredPose4" to="/$(arg uav_name)/uvdar/filteredPose4"/>

    <remap from="filteredPose1/tentative" to="/$(arg uav_name)/uvdar/filteredPose1/tentative"/>
    <remap from="filteredPose2/tentative" to="/$(arg uav_name)/uvdar/filteredPose2/tentative"/>
    <remap from="filteredPose3/tentative" to="/$(arg uav_name)/uvdar/filteredPose3/tentative"/>
    <remap from="filteredPose4/tentative" to="/$(arg uav_name)/uvdar/filteredPose4/tentative"/>


    <remap from="targetSeenCount" to="/$(arg uav_name)/uvdar/targetSeenCount"/>

  </node>
    </group>


  <!-- </group> -->



  </launch>
